rule_parser.go

```diff
@@ -14,17 +14,17 @@ type Rule struct {
 	Logsource Logsource
 	Detection Detection
 
-	ID          string        `yaml:",omitempty"`
-	Related     []RelatedRule `yaml:",omitempty"`
-	Status      string        `yaml:",omitempty"`
-	Description string        `yaml:",omitempty"`
-	Author      string        `yaml:",omitempty"`
-	Level       string        `yaml:",omitempty"`
-	References  []string      `yaml:",omitempty"`
-	Tags        []string      `yaml:",omitempty"`
+	ID          string        `yaml:",omitempty" json:",omitempty"`
+	Related     []RelatedRule `yaml:",omitempty" json:",omitempty"`
+	Status      string        `yaml:",omitempty" json:",omitempty"`
+	Description string        `yaml:",omitempty" json:",omitempty"`
+	Author      string        `yaml:",omitempty" json:",omitempty"`
+	Level       string        `yaml:",omitempty" json:",omitempty"`
+	References  []string      `yaml:",omitempty" json:",omitempty"`
+	Tags        []string      `yaml:",omitempty" json:",omitempty"`
 
 	// Any non-standard fields will end up in here
-	AdditionalFields map[string]interface{} `yaml:",inline"`
+	AdditionalFields map[string]interface{} `yaml:",inline,omitempty" json:",inline,omitempty"`
```

Comment:
Unfortunately I don't think the "inline" keyword works for json (in the yaml package used here it outputs the values at the same level of nesting as the other struct fields)

What's the use case for json output? YAML output makes sense here so that you can programmatically edit rules and then marshal them out to valid Sigma rules again

Reply:
Velociraptor emits all results in Json and we want to include the rule itself as part of the hit. Without the Json tags we get all the empty fields which just take up space.

Reply:
Makes sense 👍🏻

Reply:
Though I do wonder if you'd get better results converting YAML to JSON yourself e.g. like this https://github.com/ghodss/yaml/blob/d8423dcdf3440d0a5baffc6f90a11e4128545620/yaml.go#L104
That way you'd get support for things like `inline`

Anyway, doesn't hurt to have these tags so I'll merge